### ISSUE TITLE
Multiple Series not working

### ISSUE THREAD

**Issue by kaigame** [bigretromike/nakamoriplugin#65]
Having issues with Naruto Shippuden, .Hack//Sign, where Series does not open.

`02:26:44 T:7424   ERROR: CCurlFile::Open failed with code 404 for http://10.10.10.19:8111/JMMServerREST/GetImage/7/786/
02:26:44 T:7424   ERROR: XFILE::CHTTPDirectory::GetDirectory - Unable to get http directory (http://10.10.10.19:8111/JMMServerREST/GetImage/7/786/)
02:26:44 T:7424   ERROR: XFILE::CDirectory::GetDirectory - Error getting http://10.10.10.19:8111/JMMServerREST/GetImage/7/786/
02:26:51 T:10880   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: invalid literal for int() with base 10: ''
                                            Traceback (most recent call last):
                                              File "C:\Users\Raymond\AppData\Roaming\Kodi\addons\plugin.video.nakamoriplugin-1.2.9\nakamori.py", line 658, in <module>
                                                buildTVEpisodes(parameters)
                                              File "C:\Users\Raymond\AppData\Roaming\Kodi\addons\plugin.video.nakamoriplugin-1.2.9\nakamori.py", line 402, in buildTVEpisodes
                                                duration=int(atype.find('Media').get('duration'))/1000
                                            ValueError: invalid literal for int() with base 10: ''
                                            -->End of Python script error report<--
02:26:51 T:10356   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.nakamoriplugin/?mode=6&url=http%3a%2f%2f10.10.10.19%3a8111%2fJMMServerKodi%2fGetMetadata%2f1%2f3%2f1_92
02:26:51 T:10356   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.nakamoriplugin/?mode=6&url=http%3a%2f%2f10.10.10.19%3a8111%2fJMMServerKodi%2fGetMetadata%2f1%2f3%2f1_92) failed
02:26:53 T:10356   ERROR: Unable to activate the previous window
`

**Comment by bigretromike** [comment]
Hello, there.
I see you have missing image http://10.10.10.19:8111/JMMServerREST/GetImage/7/786/ which don't exist.
Also I see there is a error thrown from calculating duration of episode: buildTVEpisodes duration=int(atype.find('Media').get('duration'))/1000

Could you delete those two series from your JMM instalation and readd them ?

Also could you provide infromation about which group release you use, so I could also debug this on my side when readding them wont help you ?

**Comment by kaigame** [comment]
Hi, tried deleting and re-adding and the same issue, 
I'm not sure why it's labelled JMMServerREST, is this intentional? shouldn't the text say jmmserver?
The series and releases are as follows:
Naruto Shippuden: mainly Horriblesubs
.Hack//Sign: Anime HQ
One Piece: Kaizoku-Fansubs

**Comment by bigretromike** [comment]
Yes, its intentional. The way JMM works :) could you post your jmm server
version?

**Comment by kaigame** [comment]
JMMServer version is 3.4.6.4. 
Still complaining about image paths, I don't where that location corresponds to on jmmserver, as far as i can tell the folder names do not correspond to the url.

**Comment by kaigame** [comment]
Still getting errors on .Hack//Sign, DB is still rebuilding but other series appear to work, i think this will sort out the GetImage issues, but not so sure about this one:

`11:45:48 T:6252   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: invalid literal for int() with base 10: ''
                                            Traceback (most recent call last):
                                              File "C:\Users\Raymond\AppData\Roaming\Kodi\addons\plugin.video.nakamoriplugin-1.2.9\nakamori.py", line 658, in <module>
                                                buildTVEpisodes(parameters)
                                              File "C:\Users\Raymond\AppData\Roaming\Kodi\addons\plugin.video.nakamoriplugin-1.2.9\nakamori.py", line 402, in buildTVEpisodes
                                                duration=int(atype.find('Media').get('duration'))/1000
                                            ValueError: invalid literal for int() with base 10: ''
                                            -->End of Python script error report<--
11:45:48 T:4220   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.nakamoriplugin/?mode=6&url=http%3a%2f%2f10.10.10.19%3a8111%2fJMMServerKodi%2fGetMetadata%2f1%2f3%2f1_58
11:45:48 T:4220   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.nakamoriplugin/?mode=6&url=http%3a%2f%2f10.10.10.19%3a8111%2fJMMServerKodi%2fGetMetadata%2f1%2f3%2f1_58) failed`

Can play the files fine within Kodi though

**Comment by kaigame** [comment]
More findings, after some of the episodes wouldn't scrape, i have manually added them, Naruto no longer opens, prior to adding the episodes manually it worked ok.
.Hack//Sign still doesn't work, and now a few others. (one piece, project arms)

**Comment by bigretromike** [comment]
Let's debug this because its looks like JMM Database error or linking/grouping problem. I tested newest build with hack//sign and it working alright.

If you are using JMMServer on same machine type this into browser (if on other change IP to your server ip)
```http://127.0.0.1:8111/JMMServerKodi/GetMetadata/1/0/-999```
Next find (Ctrl+F) the series that dont work for you in Nakamori and copy the ```key``` argument, It should look similar to the one you typed in ex ```http://127.0.0.1:8111/JMMServerKodi/GetMetadata/1/3/207```
Open that ```key``` url in browser, save the xml (File - > Save page) and include it here

Also are you using Kodi 15 or 16?

**Comment by kaigame** [comment]
[58.txt](https://github.com/bigretromike/nakamoriplugin/files/152756/58.txt)
.Hack//Sign
[116.txt](https://github.com/bigretromike/nakamoriplugin/files/152760/116.txt)
This is Naruto Shippuden, it's massive! This is why it takes so long to load this series!

I'm running V16, let me know if there is anything else you want to see. If it helps, .Hack//Sign has sub groups for Episodes and specials, although other series are like that and work fine.

**Comment by bigretromike** [comment]
This is where the problem ocurs:
hack SIGN - 05 - Captured [AHQ].ogm JMM thinks that its duration is 0 !!! why ? not sure.
Naruto Shippuden - 392 - Episode 392.mkv same goes for this one.

I will release super quick fix for that ;-) on nakamori end. 1.3.0 Update from repo in 1h

**Comment by kaigame** [comment]
1.3.0 still has the same issue by the looks of it, although it appears as though the error is the same.
also background art seems to be a little jumbled now.

**Comment by bigretromike** [comment]
Xml's would be awesome

**Comment by kaigame** [comment]
[192.txt](https://github.com/bigretromike/nakamoriplugin/files/153465/192.txt)
Here's one for you, Project Arms, duration="" starts from episode 5 and goes all the way down :(
This series was manually added, but others were also and work ok such as Devil May Cry.

**Comment by kaigame** [comment]
have done some digging on the problematic files, they fail mediainfo checks, which is why they are showing up blank. I haven't noticed as they play fine, but if they are queried they don't show time info.

I'll repair or replace, they seem a bit bugged, sorry for the trouble.

**Comment by bigretromike** [comment]
I would love to fix this also from my end :)

**Comment by bigretromike** [comment]
Could you post your kodi.log before readding them ?:-)

**Comment by bigretromike** [comment]
1. Good news. I found myself also 'duration = 0' files in JMM Database.
2. It realy throw error still in 1.3.0 :+1: which is good for fixing it!
3. I see the problem, problem is that even duration is empty it is still parsed that way !
4. Im really sorry for semi-fixing it before, because I was unable to find same debug-sample then and I was unable to verify this behavior.
5. 1.3.1 should kill the problem in the root ;-)
6. Fun fact: few of 'duration = 0' in database use cross-ref so they have duration from other places thats why I couldn't test it in the first place

Also I added this issue to the Root of the problem https://github.com/japanesemediamanager/jmmserver/issues/226
